extensions/amp-form/0.1/amp-form.js

```diff
@@ -88,11 +92,14 @@ export class AmpForm {
         method: this.method_,
         credentials: 'include',
         requireAmpResponseSourceOrigin: true,
-      }).then(() => this.setState_(FormState_.SUBMIT_SUCCESS))
-          .catch(error => {
-            this.setState_(FormState_.SUBMIT_ERROR);
-            rethrowAsync('Form submission failed:', error);
-          });
+      }).then(response => {
+        this.setState_(FormState_.SUBMIT_SUCCESS);
+        this.renderTemplate_(FormState_.SUBMIT_SUCCESS, response);
+      }).catch(error => {
+        this.setState_(FormState_.SUBMIT_ERROR);
+        this.renderTemplate_(FormState_.SUBMIT_ERROR, error);
```

Comment:
I think it'd not be very typical to expect the error response to return JSON. But we can certainly try to parse it and include it in the rejection of the promise. E.g. `error.responseJson` or such. Then we can use it here. But I don't think it'd be a real-life usecase. Most likely one would return a successful structure that would describe erroneous situation and use template to show it as such. E.g.

```
<div submit-success>
  <template>
   {#error}
     show error
  {/#error}
  otherwise show success message
```

This case we already support with this code. If you'd like to also support the JSON in the error response - I'm fine if we do that as well. Up to you. But I don't think it'd be super common.